### PR TITLE
audio: drc: fix missing header

### PR DESCRIPTION
**Pull request by cujomalainey**

### CODE REVIEW

File: src/include/sof/audio/drc/drc.h [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 #include <stdint.h>
 #include <sof/audio/drc/drc_plat_conf.h>
+#include <sof/audio/buffer.h>
```
Comment (btian1) [inline src/include/sof/audio/drc/drc.h:12]: https://github.com/thesofproject/sof/blob/main/src/audio/drc/drc.c#L9C30-L9C30

drc.c already include buffer.h, what's the purpose to add here? if needed add here, should above be deleted?

Comment (cujomalainey) [reply src/include/sof/audio/drc/drc.h:12]: Line 61 doesn't compile without this

**Review by lgirdwood — APPROVED** [review]

File: src/include/sof/audio/drc/drc.h (hunk 1 above)
Comment (cujomalainey) [reply src/include/sof/audio/drc/drc.h:12]: Removed from the c file

Comment (btian1) [reply src/include/sof/audio/drc/drc.h:12]: const struct audio_stream __sparse_cache *source, ? this line, if this is the case, maybe simply add:
struct audio_stream;
before line 60 is fine.

Comment (cujomalainey) [reply src/include/sof/audio/drc/drc.h:12]: I would rather just include the proper header unless you have a reason not to

File: src/audio/drc/drc.c [hunk 2]
```diff
@@ -6,7 +6,6 @@
 
 #include <sof/audio/component.h>
 #include <sof/audio/data_blob.h>
-#include <sof/audio/buffer.h>
```
Comment (lyakh) [inline src/audio/drc/drc.c:9]: why are you removing it from here? Are you not using types from in in *this* file? Recursive header inclusion isn't enough, headers should be explicitly included where used. Including in both the .c and the .h is ok

File: src/include/sof/audio/drc/drc.h (hunk 1 above)
Comment (lyakh) [reply src/include/sof/audio/drc/drc.h:12]: here actually just forward declarations *might* be enough

File: src/audio/drc/drc.c (hunk 2 above)
Comment (kv2019i) [reply src/audio/drc/drc.c:9]: @lyakh isn't a bit too much to include both in foo.h and foo.c ? we are not requiring this systematically.

File: src/include/sof/audio/drc/drc.h (hunk 1 above)
Comment (kv2019i) [reply src/include/sof/audio/drc/drc.h:12]: I think including buffer.h here is ok. Forward declarations make sense in interface headers that are included in a large number of files, but I don't think we need to go overboard with this in files like drc.h that are only included in a handful of places.

**Review by kv2019i — APPROVED** [review]

**Pull request merged**